sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -217,16 +227,23 @@ private void deleteCompactedDeltas(ByteBuffer rowKey, WriteConsistency consisten
         // delete the old deltas & compaction records
         if (_writeToLegacyDeltaTable) {
             BatchStatement legacyTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
-            for (UUID change : changesToDelete) {
-                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change, consistencyLevel));
+            for (DeltaClusteringKey change : changesToDelete) {
+                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change.getChangeId(), consistencyLevel));
             }
             legacyTableFuture = session.executeAsync(legacyTableStatement);
         }
 
         if (_writeToBlockedDeltaTable) {
             BatchStatement newTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
-            for (UUID change : changesToDelete) {
-                newTableStatement.add(deleteStatement(placement.getBlockedDeltaTableDDL(), rowKey, change, consistencyLevel));
+            for (DeltaClusteringKey change : changesToDelete) {
+                if (change.hasNumBlocks() && change.getNumBlocks() < 2) {
```

Comment:
Also, if it's quick enough, we could add configs to toggle this behavior - only range vs range+cell. Also, 2nd config for the "int" number for the cell tombstone case.

Reply:
good idea